src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/ConditionMapper.java

```diff
@@ -167,14 +170,11 @@ static String getNextFieldPlaceholder(Map<String, String> expressionAttributeNam
         if (conditionExpression == null) {
             return Optional.empty();
         }
-        String toFind = keyFieldName + " = ";
-        int start = conditionExpression.indexOf(toFind);
-        if (start == -1) {
-            return Optional.empty();
-        }
-        int end = conditionExpression.indexOf(" ", start + toFind.length()); // -1
-        return Optional.of(conditionExpression.substring(start + toFind.length(),
-            end == -1 ? conditionExpression.length() : end)); // ":value" // TODO add support for non-EQ operators
+
+        // TODO add support for non-EQ operators
+        final Pattern p = Pattern.compile(keyFieldName + " = ([^ ,]*)");
+        final Matcher m = p.matcher(conditionExpression);
+        return m.find() ? Optional.of(m.group(1)) : Optional.empty();
```

Comment:
takes the first match found—if there happen to be more than one match, is that what we want here? (do we want to reject (as malformed, e.g.) if there is more than one match on keyFieldName?)

Reply:
The goal is to find the first delimiter in the string and pull the value out that comes before that delimiter.  For a "set" clause, that could be either a space, a comma, or nothing.